templates/skeleton/app/entry.server.tsx

```diff
@@ -10,7 +10,11 @@ export default async function handleRequest(
   responseHeaders: Headers,
   remixContext: EntryContext,
 ) {
-  const {nonce, header, NonceProvider} = createContentSecurityPolicy();
+  const {nonce, header, NonceProvider} = createContentSecurityPolicy({
+    connectSrc: [
+      'wss://<your-ngrok-domain.app:*', // Change with your ngrok domain
+    ],
+  });
```

Comment:
Maybe we can make it so you put in the domain during `npm run dev --ngrok-domain=accepted-shrimp-privately.ngrok-free.app` 

this will automatically start ngrok for you, do the domain change and change dev up message too.

it will mean that our `shopify hydrogen dev` will depend on ngrok. Lets discuss in a separate PR